flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskSubmissionTestEnvironment.java

```diff
@@ -167,6 +157,21 @@ private TaskSubmissionTestEnvironment(
 
 		taskExecutor.start();
 		taskExecutor.waitUntilStarted();
+
```

Comment:
I'm not sure about this.

`taskManagerServices` is used for creating the `taskExecutor`, which itself should guarantee to always access `taskSlotTable` in the RPC main thread. If there's something wrong that `taskExecutor` doesn't guarantee this, it would be better to fail the tests. Using `threadSafeTaskSlotTable` here might cover such problems.

Reply:
After checking the code, i think you are right. `TaskExecutor` is responsible for always accessing the `taskSlotTable` in the RPC main thread. We do not need to pass a `threadSafeTaskSlotTable` when building `taskManagerServices`.